src/data/nat/factorization.lean

```diff
@@ -196,4 +196,13 @@ begin
   { rintro ⟨c, rfl⟩, rw factorization_mul hd (right_ne_zero_of_mul hn), simp },
 end
 
+lemma prod_prime_factors_dvd (n : ℕ) : (∏ (p : ℕ) in n.factors.to_finset, p) ∣ n :=
```

Comment:
If you can prove the other lemma for a `multiset`, then you can prove the product over a `S.to_finset` divides the product over `S` (proving some general lemma about the product of over a submultiset) and then this will be a one-liner.

Reply:
`data/nat/factorization` wasn't compiling locally with the new lemmas added to `algebra/big_operators/` — too much stuff in between needs to re-compile, I guess — so I moved them to a new PR #11693.  When that's compiled remotely I'll merge it and re-write `prod_prime_factors_dvd` as you suggested.

Reply:
In similar cases you can also wait for CI to compile mathlib for you, and then use `leanproject get-mathlib-cache` to get the oleans, but opening a new PR is fine.

Reply:
Thanks. At the time it seemed like I might end up proving a few further lemmas about multisets, and so it would make sense to put them all together in a separate PR.